### PR TITLE
Intergration test that spans multiple epochs

### LINKED ISSUE #1195: implement integration test that spans multiple epochs
## Expected Behavior

- create a multi-node  integration test that spans multiple epochs and ensure that the nodes can stay synced (ie. that the randomness change from the epoch change is the same for all nodes, allowing blocks to still be verified)
- blocked by #1204, #1206
- This task is just creating the tests. Fixing any resulting issues will be created as new tasks

## Current Behavior

- no test spans multiple epochs

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by RyRy79261**
## Changes

- Added Dev RPC endpoints for EpochLength & SlotDuration
- Added a test with multiple nodes that waits for 3 epochs to pass

## Tests

```
go test ./tests/stress/ -t TestSync_MultipleEpoch
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1195

### CODE REVIEW

File: dot/rpc/modules/dev.go [hunk 1]
```diff
@@ -69,3 +73,23 @@ func (m *DevModule) Control(r *http.Request, req *[]string, res *string) error {
 	}
 	return err
 }
+
+// SlotDuration Dev RPC to return slot duration
+func (m *DevModule) SlotDuration(r *http.Request, req *EmptyRequest, res *string) error {
+	var err error
```
Comment (arijitAD) [inline dot/rpc/modules/dev.go:79]: Why do we return error?

File: dot/rpc/modules/dev.go [hunk 2]
```diff
@@ -69,3 +73,23 @@ func (m *DevModule) Control(r *http.Request, req *[]string, res *string) error {
...
+	buffer := make([]byte, 8)
+	binary.LittleEndian.PutUint64(buffer, m.blockProducerAPI.SlotDuration())
+	hexed := common.BytesToHex(buffer)
+	*res = hexed
+	return err
+}
+
+// EpochLength Dev RPC to return epoch length
+func (m *DevModule) EpochLength(r *http.Request, req *[]string, res *string) error {
+	var err error
```
Comment (arijitAD) [inline dot/rpc/modules/dev.go:89]: Same here. Why do we return error?

**Review by arijitAD — APPROVED** [review]

File: dot/rpc/modules/dev.go (hunk 1 above)
Comment (noot) [reply dot/rpc/modules/dev.go:79]: I believe all JSON-RPC funcs need to return error? correct me if I'm wrong @edwardmack

File: lib/babe/babe.go [hunk 3]
```diff
@@ -217,6 +217,16 @@ func (b *Service) Start() error {
 	return nil
 }
 
+// SlotDuration returns the current service slot duraiton
```
Comment (noot) [inline lib/babe/babe.go:220]: typo

File: lib/babe/babe.go [hunk 4]
```diff
@@ -217,6 +217,16 @@ func (b *Service) Start() error {
...
+func (b *Service) SlotDuration() uint64 {
+	return b.slotDuration
+}
+
+// EpochLength returns the current service epoch duraiton
```
Comment (noot) [inline lib/babe/babe.go:225]: same here

File: tests/stress/stress_test.go [hunk 5]
```diff
@@ -138,6 +138,36 @@ func TestSync_SingleBlockProducer(t *testing.T) {
 	}
 }
 
+func TestSync_MultipleEpoch(t *testing.T) {
+	numNodes = 3 // TODO: increase this when syncing improves
+	utils.SetLogLevel(log.LvlInfo)
+
+	// wait and start rest of nodes - if they all start at the same time the first round usually doesn't complete since
+	// all nodes vote for different blocks.
+	time.Sleep(time.Second * 5)
```
Comment (noot) [inline tests/stress/stress_test.go:147]: you can remove this

File: tests/stress/stress_test.go [hunk 6]
```diff
@@ -138,6 +138,36 @@ func TestSync_SingleBlockProducer(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Check state before wait
+
+	// Wait for epoch to pass
+	time.Sleep(time.Duration((slotDuration * epochLength) * 3))
+
+	// Just checking that everythings operating as expected
+	_, err = utils.SlotDuration(t, nodes[0])
+	require.NoError(t, err)
```
Comment (noot) [inline tests/stress/stress_test.go:168]: this doesn't check that the nodes stayed synced - you need to use `compareBlocksByNumber` to make sure each node has the same block hash at each number

File: tests/utils/dev.go [hunk 7]
```diff
@@ -17,11 +17,39 @@
...
 }
+
+// SlotDuration Calls dev endpoint for slot duration
+func SlotDuration(t *testing.T, node *Node) (uint64, error) {
+	slotDuration, err := PostRPC("dev_slotDuration", NewEndpoint(node.RPCPort), "")
+
+	if err != nil {
+		return 0, err
+	}
+
```
Comment (noot) [inline tests/utils/dev.go:40]: you can do `require.NoError(t, err)` in this func, then return just uint64

File: dot/rpc/modules/dev.go (hunk 1 above)
Comment (edwardmack) [reply dot/rpc/modules/dev.go:79]: Yes, that's correct.

File: tests/stress/stress_test.go (hunk 6 above)
Comment (RyRy79261) [reply tests/stress/stress_test.go:168]: Ahh rad was wondering what I could use

**Review by anthdm — APPROVED** [review]
LGTM

**Review by noot — COMMENTED** [review]
great stuff! please add unit tests for `dot/rpc/modules/dev.go` `SlotDuration` and `EpochLength`

**Pull request merged**